optimade/filterparser/lark_parser.py

```diff
@@ -96,7 +99,7 @@ def parse(self, filter_: str) -> Tree:
         """
         try:
             self.tree = self.lark.parse(filter_)
-            self.filter = filter_
+            self.filter = filter_  # type: ignore[assignment]
```

Comment:
For me   # type: ignore[assignment] is not needed anymore after you defined the type of self.filter on line 85